### ISSUE TITLE
Assert Failed on mktime call (IDFGH-9662)

### ISSUE THREAD

**Issue by pythogras** [espressif/esp-idf#11004]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

v5.0.1

### Operating System used.

Linux

### How did you build your project?

Command line with idf.py

### If you are using Windows, please specify command line type.

None

### Development Kit.

Custom board with module ESP32-WROOM-32E

### Power Supply used.

External 5V

### What is the expected behavior?

not crash on a call to mktime()

### What is the actual behavior?

Gets an assert failure on a system call to mktime()

### Steps to reproduce.

It's random and happens infrequently

### Debug Logs.

```plain
assert failed: xQueueGenericSend queue.c:837 (pxQueue->pcHead != ((void *)0) || pxQueue->u.xSemaphore.xMutexHolder == ((void *)0) || pxQueue->u.xSemaphore.xMutexHolder == xTaskGetCurrentTaskHandle())


Backtrace: 0x400820aa:0x3ffdd520 0x4008a761:0x3ffdd540 0x400919ea:0x3ffdd560 0x4008afb9:0x3ffdd680 0x40082ffb:0x3ffdd6c0 0x400831ed:0x3ffdd6f0 0x40163469:0x3ffdd720 0x4015bfb9:0x3ffdd740 0x400e21be:0x3ffdd780 0x400e2275:0x3ffdd810 0x400e22d3:0x3ffdd920 0x400e239a:0x3ffdd950 0x400e24b7:0x3ffddaf0 0x400d9d15:0x3ffddbd0 0x400e1ddc:0x3ffddbf0 0x400e1e15:0x3ffddd70 0x400e1f05:0x3ffdddf0 0x401116cb:0x3ffddfb0 0x4010f4cd:0x3ffddff0 0x40110918:0x3ffde0b0 0x4010e081:0x3ffde0d0 0x40170f3d:0x3ffde100 0x4010e353:0x3ffde120
0x400820aa: panic_abort at /opt/esp/esp-idf-5.0.1/esp-idf/components/esp_system/panic.c:423

0x4008a761: esp_system_abort at /opt/esp/esp-idf-5.0.1/esp-idf/components/esp_system/esp_system.c:153

0x400919ea: __assert_func at /opt/esp/esp-idf-5.0.1/esp-idf/components/newlib/assert.c:78

0x4008afb9: xQueueGenericSend at /opt/esp/esp-idf-5.0.1/esp-idf/components/freertos/FreeRTOS-Kernel/queue.c:837 (discriminator 8)

0x40082ffb: lock_release_generic at /opt/esp/esp-idf-5.0.1/esp-idf/components/newlib/locks.c:192

0x400831ed: _lock_release at /opt/esp/esp-idf-5.0.1/esp-idf/components/newlib/locks.c:198
 (inlined by) __retarget_lock_release at /opt/esp/esp-idf-5.0.1/esp-idf/components/newlib/locks.c:334

0x40163469: __tz_unlock at /builds/idf/crosstool-NG/.build/xtensa-esp32-elf/src/newlib/newlib/libc/time/tzlock.c:50

0x4015bfb9: mktime at /builds/idf/crosstool-NG/.build/xtensa-esp32-elf/src/newlib/newlib/libc/time/mktime.c:272

0x400e21be: gd_convertDateStrToUnixTimestamp at /home/mercieca/projects/Abertax/HEDL/Firmware_ABX043/ESP32/trunk/main/graphdata.c:710

0x400e2275: gd_processSampleLineWithDateStr at /home/mercieca/projects/Abertax/HEDL/Firmware_ABX043/ESP32/trunk/main/graphdata.c:316
```

**Comment by KaeLL** [comment]
Should be fixed by #10649, I think.

**Comment by pythogras** [comment]
Yes, its the same. 
Thanks
